submissions/crealime/memory-pair-game/js/main.js

```diff
@@ -0,0 +1,267 @@
+    pair.forEach(cardInPair => {
+      cardInPair.classList.toggle('flip')
+    })
+    pair.splice(0, 2)
+  }
+  else if (pair.length === 1 && pair[0].dataset.id === id) {
+    card.classList.toggle('flip')
+    pair.splice(0, 2)
+    pairCounter--
+    setTimeout(() => {
+      document.querySelectorAll(`.id-${id}`).forEach(cardDOMElement => {
+        cardDOMElement.classList.add('hidden')
+      })
+      if (pairCounter === 0) {
+        e.target.closest('.game').removeEventListener('click', flipCard)
+
+        gameTime.setEndTime()
+        gameTime.setDiffTime()
+        setTimeInLocaleStorage()
+        fadeOutFadeIn(initResultPage)
+      }
+    }, TIMER.time)
+  }
+  else if (pair.length === 1) {
+    card.classList.toggle('flip')
+    pair.push(card)
+  }
+  else {
+    pair.forEach(cardDOMElement => {
+      cardDOMElement.classList.toggle('flip')
+    })
+    card.classList.toggle('flip')
+    pair.splice(0, 2)
+    pair.push(card)
+  }
+}
+
+function renderCards(data, num, preview) {
+  main.innerHTML =''
+
+  const game = document.createElement('div')
+
+  game.classList.add('game')
+  game.classList.add(`num-${num * 2}`)
+
+  let cardsData = data.sort(() => 0.5 - Math.random()).slice(0, num).map((card, i) => [card, i + 1])
+
+  cardsData = [...cardsData, ...cardsData].sort(() => 0.5 - Math.random())
+
+  game.innerHTML = cardsData.reduce((acc, card) => {
+    acc += getCardTemplate(card)
+    return acc
+  }, '')
+
+  game.addEventListener('click', flipCard)
+
+  main.appendChild(game)
+
+  if (preview) showPreviewCards()
+}
+
+function initStartPage() {
+  main.innerHTML = getStartPageTemplate()
+
+  const startPageUl = document.querySelector('.start-page__ul')
+
+  startPageUl.addEventListener('click', switchFromStartPage)
+}
+
+function switchFromStartPage (e) {
+  e.preventDefault()
+
+  const link = e.target.closest('.start-page__link') || null
+  const isPreview = document.querySelector('.start-page__checkbox').checked
+
+  if (link) {
+    e.target.closest('.start-page__ul').removeEventListener('click', switchFromStartPage)
+
+    const numberOfPairs = link.dataset.pair
+    fadeOutFadeIn(initGame, numberOfPairs, isPreview)
+  }
+}
+
+function initGame(num, preview) {
+  fetch(URL).then(response => {
+    if (!response.ok) {
+      return Promise.reject(response.statusText)
+    }
+    return response.json()
+  })
+    .then(data => {
+      gameTime.setStartTime()
+      pairCounter = num
+      numOfPairs = num
+      renderCards(data, num, preview)
+    })
+    .catch(error => console.log(error))
+}
+
+function initResultPage() {
+  main.innerHTML = getResultPageTemplate(gameTime.getDiffTime(), numOfPairs)
+
+  const resultPageLink = document.querySelector('.result-page__link')
+
+  resultPageLink.addEventListener('click', switchFromResultPage)
+}
+
+function switchFromResultPage (e) {
+  e.preventDefault()
+  e.target.removeEventListener('click', switchFromResultPage)
```

Comment:
`e.target` not always equal to `resultPageLink` [Read more about that](https://medium.com/@etherealm/currenttarget-vs-target-in-js-2f3fd3a543e5)

Reply:
Done.